calyx/src/analysis/domination_analysis/node_analysis.rs

```diff
@@ -170,15 +170,73 @@ impl NodeSearch {
         NodeSearch { name }
     }
 
+    // 1) checks whether assign is of form dst.go = src.done;
+    // 2) checks whether assignments contains both:
+    // src.go = 1;
+    // group[done] = dst.done;
+    // If all of this is true, then returns true: we can be confident dst is
+    // written to. otherwise return false
+    fn special_case(
```

Comment:
More informative name needed here. Also, the comment doesn't define the relationship between `src` and `dst`